### ISSUE TITLE
Shorter URLs

### ISSUE THREAD

**Issue by dstufft** [pypi/inspector#8]
Currently URLs look something like this:

```
https://inspector.pypi.io/project/pip/22.1/packages/f3/77/23152f90de45957b59591c34dcb39b78194eb67d088d4f8799e9aa9726c4/pip-22.1-py3-none-any.whl/pip/_internal/models/format_control.py
```

That's pretty long!  Obviously this is done because that gives enough information to fetch the URL from ``files.pythonhosted.org``, but it might be nice to use shorter URLs, and query PyPI to get the long URL for the file distribution?

We could go as simple as:

```
https://inspector.pypi.io/file/pip/pip-22.1-py3-none-any.whl/pip/_internal/models/format_control.py
```

That's enough information to know the project name (since sdists don't have a well formed name) and the filename (which we can then look up on PyPI's ``/simple/<project>/`` page), and get the long URL.

We could even go a bit simpler, and do:

```
https://inspector.pypi.io/file/pip-22.1-py3-none-any.whl/pip/_internal/models/format_control.py
```

Note all this embeds is the filename, we would need a way to look up the URL given nothing but the filename, but filenames *are* unique on PyPI, so we could just have a route on PyPI that does a redirect of filename to pythonhosted.org and does that look up for us.

The main thing we'd lose is that these links would then "die" if the file is removed from PyPI but still exists in files.pythonhosted.org. Maybe with https://github.com/pypi/inspector/issues/5 we could store the filename => file url mapping as we load them, which would mean they would continue to work in the future.

Alternatively, maybe still support the long URLs, and have a button to turn the short url into a permalink (think how github does).

Alternatively, maybe this is a silly idea and we should just stick with the long URLs :)

**Comment by di** [comment]
> Note all this embeds is the filename, we would need a way to look up the URL given nothing but the filename, but filenames are unique on PyPI, so we could just have a route on PyPI that does a redirect of filename to pythonhosted.org and does that look up for us.

Yeah, this is why I did this, so that these links continue to work after the project is taken down, without having to make changes to PyPI.
